### PR TITLE
docs: Docs Version Dropdown Automation

### PR DESCRIPTION
**Pull request by MikhailKardash**
## Description

This script is step 1 of 2 for version dropdown automation. It inserts a version entry into the versions-switcher json. 

Step 2 is calling this script during the release process. Those scripts are in a separate repo.

## Commentary

Perhaps we should start a `docs_tools` subdirectory within `docs`. As a python dev I expect `docs/tools` to be something different from what it currently is in the repo, maybe we should refactor?

## Test Plan

1. `cd docs`
2. `python insert-version-url.py test`
3. Open `versions.json` and see that there's a `test` entry in the second position.

## Checklist

- [X] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
MLG-888

### CODE REVIEW

**Review by rb-determined-ai — APPROVED** [review]

**Review by rb-determined-ai — APPROVED** [review]

File: docs/insert-version-url.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+def create_url_entry(version):
+    url = f"https://docs.determined.ai/{version}/"
+
+    entry = {"version": version, "url": url}
+
+    return entry
+
+
+def insert_entry(json_file, entry, entry_position=1):
```
Comment (dzhu) [inline docs/insert-version-url.py:22]: Having `entry_position` as an argument that's never used seems like overkill for this tiny script, and the value could bear to be explained.
```suggestion
def insert_entry(json_file, entry):
    # The current dev version comes first and all released versions are in revers chronological
    # order, so the latest release always goes in index 1.
    entry_position = 1
```

File: docs/insert-version-url.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    all_urls.insert(entry_position, entry)
+
+    with open(json_file, "w") as f:
+        json.dump(all_urls, f, indent=4 * " ")
+        f.write("\n")
+
+
+def insert_version_url(json_file, version):
+    new_entry = create_url_entry(version)
+    insert_entry(json_file, new_entry)
```
Comment (dzhu) [inline docs/insert-version-url.py:35]: I don't know that we really need a whole other six-line function to create a dict with two entries.
```suggestion
    insert_entry(json_file, {"version": version, "url": f"https://docs.determined.ai/{version}/"})
```

File: tools/scripts/insert-dropdown-url.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash -ex
+
+# check for dirty changes
+if [[ -n "$(git status --porcelain)" ]]; then
+    echo "untracked or dirty files are not allowed, cleanup before running insert-dropdown-url.sh"
+    exit 1
+fi
+
+# insert-version-url.py inserts a new entry into the versions.json--adding a dropdown link
+python3 docs/insert-version-url.py $(grep -oE '^[0-9.]*' VERSION)
```
Comment (dzhu) [inline tools/scripts/insert-dropdown-url.sh:10]:
```suggestion
python3 docs/insert-version-url.py "$(grep -oE '^[0-9.]*' VERSION)"
```
Doesn't make a difference here, but prefer not to have to think about whether it does.

**Review by dzhu — APPROVED** [review]

File: docs/insert-version-url.py (hunk 1 above)
Comment (MikhailKardash) [reply docs/insert-version-url.py]: I don't mind this. It explains the code better anyways.

**Pull request merged**